designs/sam-debugging/local-sam-debugging.md

```diff
@@ -0,0 +1,308 @@
+| Runtime                 | Runtime of Lambda Function to invoke                 | x                                 |                                |
+| Handler                 | Lambda Function Handler to invoke                    | x                                 |                                |
+| Timeout                 | Timeout threshold for Lambda function                | x                                 |                                |
+| Memory                  | Memory provided to Lambda function                   | x                                 |                                |
+
+The following SAM CLI related arguments are relevant to debugging both standalone lambda function handlers and sam template resources. For reference see the [sam build](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-build.html) command.
+
+| Property                                                         | Default Value                    |
+| ---------------------------------------------------------------- | -------------------------------- |
+| Build SAM App in container                                       | false                            |
+| Skip new image check                                             | false                            |
+| use a docker network                                             | empty string (no docker network) |
+| additional build args (passed along to `sam build` calls)        | empty string                     |
+| additional local invoke args (passed along to `sam local` calls) | empty string                     |
+
+The following AWS related arguments are relevant to debugging both standalone lambda function handlers and sam template resources:
+
+-   Credentials
+-   Region
+
+## <a id="debugging-experiences"></a> Local Debugging Experiences
+
+### Debug Configurations
+
+Debug Configurations are the idiomatic approach to running and debugging software in VS Code. They are also a reusable component - the Toolkit is able to internally produce and execute these configurations on the fly (for example as a part of the [CodeLenses](#codelenses) functionality). This is the Toolkit's main experience for debugging SAM Template resources.
+
+The Toolkit implements a Debug Configuration type `aws-sam`. Users can author and maintain these configuration entries, then launch them by pressing F5. When launched, this configuration type:
+
+-   validates debug configuration inputs
+-   uses SAM CLI to build a SAM Application
+-   uses SAM CLI to invoke a SAM Template Resource
+-   attaches a debugger to the SAM invocation
+-   if the debug configuration is for a local api gateway invoke, the debugger is detached after the http request is made, but SAM CLI remains active. The debug configuration implementation terminates the SAM CLI session to prevent a proliferation of CLI processes.
```

Comment:
ya kinda lost me here.

Reply:
Rambly. Thanks for the catch...

Revised to: 
if the Debug Configuration is for an API Gateway SAM Template Resource (link here to the section around line 51), the SAM-based http host is terminated to prevent a proliferation of CLI processes